packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -80,14 +80,12 @@ export default createConnector({
 
     this._prevState = this._prevState || {};
 
-    const cache = props.cache || getInMemoryCache();
-    if (this._cachedHits === undefined) {
-      this._cachedHits = cache.read({ state: searchState }) || {};
-    }
+    this._cache = props.cache ? props.cache : this._cache || getInMemoryCache();
```

Comment:
this is a very important fix!

Reply:
Yes, it especially makes sure to take `cache` prop into account even when user passes a different object!

Reply:
and makes sure we don't recreate the memory cache on every render